sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp

```diff
@@ -110,6 +110,8 @@ __ESIMD_INTRIN __ESIMD_raw_vec_t(T, N)
 }
 #endif // __SYCL_DEVICE_ONLY__
 
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
The agreed way to structure intrinsics is
```
void __esimd_intrinsic(...)
#ifdef __SYCL_DEVICE_ONLY__
;
#else
{ <emu_implemenation_of_intrinsic>; }
#endif // __SYCL_DEVICE_ONLY__
```
this way we guarantee same signatures on host and device.

Reply:
I don't know why, but `__esimd_dpas*` function signatures are different per __SYCL_DEVICE_ONLY__ from caller side. For example, `dpas_info` argument does not exist for without __SYCL_DEVICE_ONLY__.

https://github.com/intel/llvm/blob/dd17a3363e2538391cfd74b3dbdaa39b13827202/sycl/include/sycl/ext/intel/experimental/esimd/math.hpp#L2025

Reply:
`__esimd_dpas*` functions are changed to have same function signature regardless of `__SYCL_DEVICE_ONLY__` and their definitions are changed as suggested.

Previously, `src1_precision`, `src2_precision`, `systolic_depth`, and `repeat_count` are template arguments for emulation while they are runtime arguments for (actual) devices. I converted all of those template arguments into runtime arguments so that function signatures are same between with and without `__SYCL_DEVICE_ONLY__`.